code/datums/status_effects/debuffs.dm

```diff
@@ -180,31 +180,69 @@
 	ADD_TRAIT(owner, TRAIT_KNOCKEDOUT, TRAIT_STATUS_EFFECT(id))
 	tick_interval = initial(tick_interval)
 
+#define HEALING_SLEEP_DEFAULT -0.2
+
 /datum/status_effect/incapacitating/sleeping/tick()
 	if(owner.maxHealth)
 		var/health_ratio = owner.health / owner.maxHealth
-		var/healing = -0.2
+		var/healing = HEALING_SLEEP_DEFAULT
+
+		// having high spirits helps us recover
+		var/datum/component/mood/mood = owner.GetComponent(/datum/component/mood)
+		if(mood != null)
+			switch(mood.sanity_level)
+				if(SANITY_LEVEL_GREAT)
```

Comment:
Use these defines in the code that defined their values please

Reply:
Fixed with https://github.com/tgstation/tgstation/pull/65713/commits/77316bbf8606685e60d098046dfe3d5dce54a8c8